concepts-wip/bools/about.md

```diff
@@ -0,0 +1,144 @@
+# Bools
+
+Crystal has a type know as [`Bool`][bools], it is used to represent the values `true` and `false`.
+With the bool type can you use logical operators(`!`, `||`, `&&`) which can be used to combine bools and change their values.
+
+```crystal
+true && true
+# => True
+
+true && false
+# => False
+
+false && false
+# => false
+
+false || true
+# => True
+false || false
+# => false
+
+!false
+# => true
+
+!true
+# => false
+```
+
+## Logical operators
+
+Crystal has 3 logical operators which is used to combine bools and change their values.
+
+### And(`&&`)
+
+The [_and_ operator][and] in crystal is represented by `&&` and returns `true` if both values given is `true` otherwise it returns `false`.
+When using the _and_ operator so should one bool be placed one the right side of the `&&` and one on the left side.
+
+```crystal
+true && true
+# => true
+
+true && false
+# => false
+
+false && false
+# => false
+```
+
+### Or(`||`)
+
+The [_or_ operator][or] in crystal is represented by `||` and returns `true` if **at least one** of values given is `true` if both of the values are `false` then it returns `false`.
+When using the _or_ operator so should one bool be placed one the right side of the `||` and one on the left side.
+
+```crystal
+true || true
+# => true
+
+true || false
+# => true
+
+false || true
+# => true
+
+false || false
+# => false
+```
+
+### Not(`!`)
+
+The _not_ operator in crystal is represented by `!` and returns `true` if the given bool is `false` and return `false` if `true` is given.
+When using the _not_ operator so should the bool be placed after the operator (`!`).
+
+```crystal
+!true
+# => false
+
+!false
+# => true
+```
+
+## Using brackets(`()`)
+
+When working with booleans you can use brackets to decided which bools to evaluate first.
+Dependent on how the brackets are used so can the result end up to be different.
+
+In crystal so is what is in brackets evaluated first.
+After brackets has been evaluated so are booleans evaluated from left to right.
+In the example below so will first _and_ operation be evaluated which will evaluate to: `false && false || true`.
+Then the second _and_ operation which makes it look like: `false || true`, which results in `true`.
+In the second example it will first evaluate what is in the brackets, after the evaluation it will look like: `true && false && true`.
+In a situation where there is only _and_ operation so does all bools has to be true to make the result be true, since one is false so will the result be false.
+
+```crystal
+true && false && false || true
+# => true
+
+true && false && (false || true)
+# => false
+```
+
+Since what is in the bracket is evaluated first so will it affect when using the _not_ operator.
+
+```crystal
+!true && false
+# => false
+
+!(true && false)
+# => true
+```
+
+```exercism/caution
+You should only use brackets when they affect the result, otherwise should they be omitted.
+```
```

Comment:
I have a bit of mixed feelings about this, I have seen on exercism quite mixed use of both ways of writing. And the markdown spec says you can use backticks.

I think following the official way is great, although the use as I said is quite mixed across the track.
And some stuff I write for crystal is being very lightly reviewed (for example the docs for the generator, some stuff is being more highly reviewed and thereby would not apply to those) then it makes sense to have a formatter that can catch for example double spaces. The formatter I use does not like tildas to be used that way. And removes them.

I will have to think over this, I think tildas could be a good way of doing it, since it makes the "blocs" different from code examples in the markdown file. But I want my formatter so yeah, let me think about it.

Reply:
If backticks work, I think it's fine to use them. I was just wondering if they would work at all.

Reply:
No I think it is great you pointed this out, and yes they do work for just text.

But when wanting a code block inside of them, am I a bit unsure, 4 backticks maybe make it work. But I am pretty sure 3 won't. So in any case I will have to do a change to make it consistent across the track.